spec/models/enterprise_fee_spec.rb

```diff
@@ -9,6 +9,19 @@
 
   describe "validations" do
     it { is_expected.to validate_presence_of(:name) }
+
+    it "requires a per-item calculator to inherit tax" do
+      subject = build(
+        :enterprise_fee,
+        inherits_tax_category: true,
+        calculator: Calculator::FlatRate.new
+      )
+
+      expect(subject.save).to eq false
+      expect(subject.errors.full_messages.first).to eq(
+        "Inheriting the tax categeory requires a per-item calculator."
+      )
+    end
```

Comment:
In addition to this one, I'd suggest to add an it block, in which the validation is not triggered and saving into the DB is successful.

Reply:
`±` Good suggestion 👍

Reply:
I think we should still add a test for the successful cases. I think the most important one is:

* Per-item calculator, with inheritance ticked

Reply:
I updated the tests to check the per-order and per-item calculators.
Now, I'm thinking we only discussed the two types of calculators, how about the others?
![image](https://user-images.githubusercontent.com/16113997/230706415-686e0790-98db-499a-af23-b9b0c2109d3e.png)

Reply:
I can see in one of @filipefurtad0's [comments](https://github.com/openfoodfoundation/openfoodnetwork/issues/10348#issuecomment-1410664238) that these three can inherit tax category:

> Flat percent (per item), Flat rate (per item), Weight

So I assume all others cannot. Is that right Filipe?

Reply:
if that's true, maybe we can extends the original issue to include all the categories (except per-item) ?

Reply:
We've clarified it over a huddle with @abdellani - I think we're good: weight is indeed applied over a line item, but should be included in the PR already :+1

Reply:
Ok, sounds like we're all good!